Anguandia/quickcredit
# Add route authorization

#### Description of Task to be completed?

- Add user authentication and route authorization middleware
- Add auth to routes requiring access restriction
- Add tests for authentication
- Modify existing tests to factor in the auth header

#### How should this be manually tested?

- Tests general authentication
  - Send a post/api/v1/loans request on postman without inserting the
  authorization token
  - The response should ask for sign in
- Test Admin Authorization
  - Signup with a post/api/v1/auth request and your names and email only. This creates a client user
  - Copy the token from the response and past in the authorization token space
  - Create a loan
  - Try to approve the created loan
  - The response should notify that admin required
- Test resource owner authentication
  - Sign up two separate users as before
  - Create a loan with the user as the first user's email
  - Send a get/api/v1/loans/1/repayments request(no problem if no repayment has been transacted, authentication comes before request execution
  - The response should state that the owner is required to access this resource

#### What are the relevant pivotal tracker stories?

[finishes #166027460]
